src/Sentry.Unity.Android/JniExecutor.cs

```diff
@@ -91,14 +98,21 @@ private void DoWork()
 
             try
             {
-                return (TResult?)_taskCompletionSource.Task.GetAwaiter().GetResult();
+                if (!_taskCompletionSource.Task.Wait(TimeoutMs))
+                {
+                    throw new TimeoutException($"JNI operation timed out after {TimeoutMs}ms");
+                }
+                return (TResult?)_taskCompletionSource.Task.Result;
```

Comment:
I can see that. I've updated the PR to make use of the cancellation token instead of waiting. We could link the cancellation sources later on if we wanted to.
Right now we're checking for 3 things of the Android SDK: `IsEnabled`, `GetInstallationId` and `CrashedLastRun`. Otherwise it's all fire and forget.